machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,143 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigNode describes the health of the Machines on the system
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this specific machineconfignode
+	// +kubebuilder:validation:Required
+	Spec MachineConfigNodeSpec `json:"spec"`
+
+	// status describes the last observed state of this machineconfignode
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status MachineConfigNodeStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigNodeList describes all of the MachinesStates on the system
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigNodeList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineConfigNode `json:"items"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// kind describes the type of object thsi reference points to
+	// +kubebuilder:validation:Required
+	// +required
+	Kind string `json:"kind"`
+	// name describes what this object is called
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// apiVersion details the group and version of this resource.
+	// +kubebuilder:validation:Required
+	// +required
+	APIVersion string `json:"apiVersion"`
+}
+
+// MachineConfigNodeSpec describes the type of State we are managing
+type MachineConfigNodeSpec struct {
+	// nodeRef contains references to the node for this MachineConfigNode
+	// +optional
+	NodeRef MCOObjectReference `json:"nodeRef"`
```

Comment:
True. I can do the same thing with the pool I guess since it is always a pool. In that case I can just have them be strings rather than a whole object type.

Reply:
oh well I guess if you want name nested inside of node it needs to be an object.